tests/functional/testSubscriptionNeeded.js

```diff
@@ -105,7 +105,26 @@ describe("subscription needed tests", function() {
             eventName: "alreadyASubscriberSelected",
             screen
         });
+      })
+
+      it("successfull outcome events are recorded", async () => {
+        // Click the "Subscribe now" button
+        await vpn.waitForQueryAndClick(queries.screenSubscriptionNeeded.SUBSCRIPTION_NEEDED_BUTTON.visible());
+        const startedEventsList = await vpn.gleanTestGetValue("outcome", "subscriptionStarted", "main");
+        assert.strictEqual(startedEventsList.length, 1);
+
+        // Subscription, for Guardian, is the same as in-browser auth.
+        await vpn.mockInBrowserAuthentication();
+
+        const completedEventsList = await vpn.gleanTestGetValue("outcome", "subscriptionCompleted", "main");
+        assert.strictEqual(completedEventsList.length, 1);
```

Comment:
This is not working and also this is missing tests for failed states.

Reply:
As commented on https://github.com/mozilla-mobile/mozilla-vpn-client/pull/9150#issuecomment-2090057667 failure events are out of scope for this ticket.